RubeesEat/Model/DB/DbBillRepository.cs

```diff
@@ -162,15 +162,14 @@ public decimal GetBalance(Person user)
 
         connection.Open();
 
-        using var reader = command.ExecuteReader();
-
-        decimal sum = 0;
-        while (reader.Read())
+        var result = command.ExecuteScalar();
+        
+        if (result == DBNull.Value)
         {
-            sum += reader.GetDecimal(0);
+            return 0m;
         }
 
-        return sum;
+        return Convert.ToDecimal(result);
```

Comment:
Why do we need this change?

Reply:
This is a bug that has been here for a while but wasn't noticed. When the current user has no bills, and the website wants to display the balance, the sql statement SUM then returns Null, instead of 0, and the function breaks when trying to convert to decimal

When a users login name is not in the db, a new Person is created for them, with no bills yet, and this bug shows up. This is why it was noticed in this branch.

So i fixed the bug here in a seperate commit. Now the function returns 0, when the sql query returns null.

Reply:
Irem and Joel rebased on main, as preparation for merging later. Unfortunately I forgot about this commit, so I couldn't tell them to leave this commit alone.

Reply:
Is extracted: https://github.com/re-motion/students-RubeesEat/pull/44